File: registratorapp/src/main/java/de/weimarnetz/registrator/services/PasswordService.java
package de.weimarnetz.registrator.services;

import org.mindrot.jbcrypt.BCrypt;
import org.springframework.stereotype.Component;

@Component
public class PasswordService {

    private static final int LOG_ROUNDS = 8;

    public String encryptPassword(String password) {
        return BCrypt.hashpw(password, BCrypt.gensalt(LOG_ROUNDS));
    }

}

File: src/main/java/de/weimarnetz/registrator/services/PasswordService.kt
package de.weimarnetz.registrator.services

import org.mindrot.jbcrypt.BCrypt
import org.springframework.stereotype.Component

private const val LOG_ROUNDS = 8

@Component
class PasswordService {
    fun encryptPassword(password: String): String {
        return BCrypt.hashpw(password, BCrypt.gensalt(LOG_ROUNDS))
    }
}